### ISSUE TITLE
Running MNIST forward, loss and backward graph (without optimizer)

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-forge-fe#174]
## Summary
For initial run, having only forward and backward pass should be sufficient (optimizer can be introduced for full e2e training). This also includes loss function.

## Details
Initial training can work with following components:
- Forward graph + Loss attached
- Backward graph

It won't train e2e, but it's good proof of concept that we can run it in a stable manner. 

Also, instead creating training wrappers that are aware of fwd, bwd, and loss components, we can do modification on user graph to make this PoC a bit easier. E.g.:
```py
class ModuleAndLoss(nn.Module):
    def __init__(self, module, loss):
        self.module = module
        self.loss = loss

    def forward(self, inputs, targets):
        return self.loss(self.module(inputs), targets)
```

This way, loss graph is mapped and attached to the fwd graph directly (as it's currently in PyBuda/BBE stack; more details here tenstorrent/tt-forge-fe#176 ). 

## Passing criteria
- We're generating separated flatbuffers for fwd and bwd graphs (try to incorporate loss in fwd/bwd if possible)
- Golden/Silicon pcc is matching for both fwd and bwd passes
